core/server/src/main/java/alluxio/master/file/meta/MutableLockedInodePath.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.exception.InvalidPathException;
+
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class represents an {@link LockedInodePath}, where the list of inodes can be extended to
+ * gather additional inodes along the path.
+ */
+@ThreadSafe
+public final class MutableLockedInodePath extends LockedInodePath {
```

Comment:
FYI, the it seems that the precedent in the standard Java API is for the default to be mutable and the immutable object to throw UnsupportedOperationException if methods that wish to modify it are invoked: http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Collections.html#unmodifiableList(java.util.List)